hack/shared_types.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+from typing import TypedDict
+
+class RepositoryInfo(TypedDict):
+    repository: str
+    ref: str
+    image_url: str | None
+
+class SnapshotMetadata(TypedDict):
+    generated_at: str
+    status: str
+    tested_at: str
+    id: str
+
+class SnapshotInfo(TypedDict):
+    metadata: SnapshotMetadata
+    components: list[RepositoryInfo]
+
+class ReleaseCandidates(TypedDict):
+    snapshots: list[SnapshotInfo]
+
+class VersionEntry(TypedDict):
+    name: str
+    components: list[RepositoryInfo]
+
+class VersionsFile(TypedDict):
```

Comment:
What's the difference between VersionFile and VersionEntry? The way I see it is that the file is just where we save a list of released versions. I think we should not have the word `file` in the name.
I think we might rethink how we represent those. What's the relation between the handled objects? The way I see it is:
* we discover a new "version candidate" or "snapshot"
* we validate such snapshot, if valid can be promoted to versions
* once added to versions, it will be reconciled

Can we find a way to represent this?